### PR TITLE
refactor: rewrite popup syntax based on python syntax v2

### LINKED ISSUE #179: Wrong syntax highlighting for function signature
image

```py
(function) fail: _WithException[(reason: str = "", pytrace: bool = True, msg: str | None = None) -> NoReturn, Type[str]]
# ^^^^^^^ storage.type
#                ^^^^^^^^^^^^^^ meta.qualified-name meta.generic-name
#                                ^^^^^^ variable.parameter
#                                        ^^^ support.type
#                                            ^ keyword.operator.assignment
#                                              ^^ string.quoted.double
#                                                  ^^^^^^^ variable.parameter
#                                                           ^^^^ support.type
#                                                                ^ keyword.operator.assignment
#                                                                  ^^^^ constant.language.boolean
#                                                                        ^^^ variable.parameter
#                                                                             ^^^ support.type
#                                                                                 ^  keyword.operator
#                                                                                   ^^^^ constant.language.null
#                                                                                        ^ keyword.operator.assignment
#                                                                                          ^^^^ constant.language.null
#                                                                                                ^^ punctuation.separator
#                                                                                                   ^^^^^^^^ meta.qualified-name meta.generic-name
#                                                                                                                  ^^^ support.type
#                                                                                                                     ^^ punctuation.section.sequence.end
```

### LINKED ISSUE #220: Vendor Python syntax or drop ST 3
because the Python syntax is going to be v2 in https://github.com/sublimehq/Packages/pull/3650 and we extend it in https://github.com/sublimelsp/LSP-pyright/blob/master/syntaxes/pyright.sublime-syntax

### PR DESCRIPTION
**Pull request by deathaxe**
This is an attempt to rewrite popup syntax based on new Python syntax of https://github.com/sublimehq/Packages/pull/3650.

I am not too confident with `Overload[...]` syntax, but the rest should work well.

Class instantiation uses a dedicated context to apply `meta.class.parameters` scope, in order to keep consistent with `meta.class` which is used for class name.

---

Closes #179
Closes #220

### CODE REVIEW

**Comment by jfcherng** [comment]
I am not sure whether we are going to bundle v2 Python syntax when it's released with next ST build. The reason I bundled it now is to prevent from not usable if the next ST version is just released.

**Comment by jfcherng** [comment]
🤔 Weird. How can the 4126 CI test pass? The Python syntax definition in ST 4126 uses v1 while this PR uses v2.

File: syntaxes/pyright.sublime-syntax [hunk 1]
```diff
@@ -2,150 +2,167 @@
...
+  function-definition-end:
+    - meta_prepend: true
     # match trailing ending brackets in LSP popup
-    - match: \)
-      scope: punctuation.section.parameters.end.python
-      pop: true
-    - match: \]
-      scope: punctuation.section.sequence.end.python
-      pop: true
+    - match: (?=[])])
```
Comment (jfcherng) [inline syntaxes/pyright.sublime-syntax:157]: It seems working but I personally prefer escaping `]`.

```suggestion
    - match: (?=[\])])
```

**Review by jfcherng — APPROVED** [review]

**Comment by jfcherng** [comment]
Could you also remove CI test for ST 3 since it's no longer needed after this PR.

File: syntaxes/pyright.sublime-syntax (hunk 1 above)
Comment (deathaxe) [reply syntaxes/pyright.sublime-syntax:157]: And I was so proud to have saved one char 🤣 .

**Comment by jfcherng** [comment]
noticed a minor issue

image

```py
class ClassName(foo: int, /, *args: Any, **kwargs: Any)
# ^^^ keyword.declaration.class
#     ^^^^^^^^^ entity.name.class
#               ^^^ variable.parameter
#                  ^ punctuation.separator.annotation
#                    ^^^ support.type
#                         ^ storage.modifier.positional-args-only
#                            ^ keyword.operator.unpacking.sequence
#                             ^^^^ variable.parameter
#                                 ^ punctuation.separator.annotation
#                                   ^^^ meta.generic-name
#                                      ^ punctuation.separator
#                                        ^^ keyword.operator.unpacking.mapping
#                                          ^^^^^^ variable.parameter
#                                                  ^^^ meta.generic-name

(method) def functionName(cls: Type[Self@ParentName], /, key: str, default: Any | None = None) -> Any
# ^^^^^ storage.type
#                         ^^^ variable.parameter
#                                                     ^ storage.modifier.positional-args-only
#                                                        ^^^ variable.parameter
#                                                             ^^^ support.type
#                                                                  ^^^^^^^ variable.parameter
#                                                                               ^  keyword.operator
#                                                                                      ^ keyword.operator
#                                                                                              ^^ punctuation.separator


```

**Comment by jfcherng** [comment]
Thanks for your effort! There is a bug after this PR but I think we can fix it later in another PR. https://github.com/sublimelsp/LSP-pyright/issues/233

**Pull request merged**